### PR TITLE
Fix #18 Add live reload support

### LINKED ISSUE #18: Live reload
**Is your feature request related to a problem? Please describe.**
During the preparation of the presentation, I need to restart the program many times.

**Describe the solution you'd like**
Live reload helps to keep one terminal on a separate monitor running the presentation (and refresh it automatically), while editing the Markdown file on another monitor.

### PR DESCRIPTION
**Pull request by sibprogrammer**

### CODE REVIEW

**Comment by maaslalani** [comment]
Hey! Thanks so much for the contribution again! I was thinking that it probably shouldn't be the responsibility of slides to ensure that your slides are automatically reloaded. I built a tool called [crow](https://github.com/maaslalani/crow) to do something similar where it will watch files and re-execute commands when the watched files are changed, so it would be something like:

~~~
crow -w presentation.md slides presentation.md
~~~

And that would re-run slides whenever that file changed, so that slides doesn't need to be responsible for watching the files. This doesn't currently work since theres some issues with TTY stuff, but if thats an okay solution for you I would be happy to look into how I can make it all work together.

**Comment by sibprogrammer** [comment]
> if thats an okay solution for you
It will not work well because the internal state will be lost after relaunch.
My use case is quite simple: I am working on the presentation and now editing slide 4, as soon as I changed the content of this slide, I want that another terminal with opened deck to refresh it automatically and stay on slide 4. Overwise it will be quite useless.

**Comment by maaslalani** [comment]
See #18 where we would have a `SLIDES_PAGE` variable for restarting at the same page.

I'm not 100% sold either way, so I could certainly be convinced.

**Comment by maaslalani** [comment]
I think `live` reload does makes sense. Can you think of any reason why it shouldn't be on by default? I'm wondering if we should just always have it reflect live edits. Obviously this would only work with passing the file as an argument, and not reading from stdin. If users absolutely do not want `live` reload they can `cat file.md | slides`.

What do you think?

**Comment by sibprogrammer** [comment]
> Can you think of any reason why it shouldn't be on by default?

I think it's quite reasonable to enable it by default. I agree that the desire to avoid live reload looks like an edge case and it's enough to have `cat file.md | slides` as a workaround.

**Comment by maaslalani** [comment]
Sweet, then I think we can get rid of the `--live` flag and do live reloading if a filename is passed.

**Comment by sibprogrammer** [comment]
@maaslalani plz take a look

File: internal/model/model.go [hunk 1]
```diff
@@ -42,6 +101,19 @@ func (m Model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
...
+		newFileInfo, err := os.Stat(m.FileName)
+		if err == nil {
+			if newFileInfo.ModTime() != fileInfo.ModTime() {
+				fileInfo = newFileInfo
+				_ = m.Load()
+				if m.Page >= len(m.Slides) {
+					m.Page = len(m.Slides) - 1
+				}
+			}
+		}
```
Comment (maaslalani) [inline internal/model/model.go:115]: Can you fix the code to have a more flat structure, something like this?

```suggestion
		if err != nil {
			return m, fileWatchCmd()
		}
		// File unchanged, skip reload
	        if newFileInfo.ModTime() == fileInfo.ModTime() {
			return m, fileWatchCmd()
	        }
		fileInfo = newFileInfo
		_ = m.Load()
		if m.Page >= len(m.Slides) {
			m.Page = len(m.Slides) - 1
		}
```

**Comment by maaslalani** [comment]
I tested this out and it looks really good 😄

File: internal/model/model.go [hunk 2]
```diff
@@ -42,6 +101,19 @@ func (m Model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 				m.Page--
 			}
 		}
+
+	case fileWatchMsg:
+		newFileInfo, err := os.Stat(m.FileName)
+		if err == nil {
+			if newFileInfo.ModTime() != fileInfo.ModTime() {
```
Comment (maaslalani) [inline internal/model/model.go:108]: We are pretty much implementing our own polling method every one second, is there any way to use something like  https://github.com/fsnotify/fsnotify so we don't have to poll and we get instant updates.

File: internal/model/model.go (hunk 1 above)
Comment (sibprogrammer) [reply internal/model/model.go]: I don't think that repeating the line `return m, fileWatchCmd()` 3 times is a better option :)
Anyway, I reduce the level of nesting a little bit.

File: internal/model/model.go (hunk 2 above)
Comment (sibprogrammer) [reply internal/model/model.go]: I'm not so familiar with bubbletea library to simulate `tea.Msg` event and integrate seamlessly it into the render loop. On the other hand, in the case of `slides` app one stats call per second is not so expensive. Switch to inotify/fsevent seems like will give almost zero performance benefit.

Comment (maaslalani) [reply internal/model/model.go]: Sounds good, its mainly to have instant reloading rather than 1 second delayed reloading.

Comment (maaslalani) [reply internal/model/model.go]: I might have a look later on how to use fsnotify

File: internal/model/model.go (hunk 1 above)
Comment (maaslalani) [reply internal/model/model.go]: Thanks!

**Comment by maaslalani** [comment]
Great work @sibprogrammer! Thanks so much for the contribution ❤️

**Pull request merged**